### PR TITLE
fix(release): fix trigger release

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/KFLUXBUGS-1176, 
https://issues.redhat.com/browse/KFLUXBUGS-1177, 
https://issues.redhat.com/browse/KFLUXBUGS-1178

## Description

1. Pre selects release plan
2. Only lists snapshots from same application
3. redirects to newly crated Release details page.

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

ScreenRecording2024-04-08at7 22 54PM-ezgif com-video-to-gif-converter

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by scoheb** [comment]
what is the status of this PR?

File: src/pages/TriggerReleasePlanPage.tsx [hunk 1]
```diff
@@ -1,20 +1,66 @@
...
+import { HttpError } from '../shared/utils/error/http-error';
 import { AccessReviewResources } from '../types';
+import { ReleasePlanKind } from '../types/coreBuildService';
+import { useWorkspaceInfo } from '../utils/workspace-context-utils';
 
 const TriggerReleasePlanPage: React.FC<React.PropsWithChildren<unknown>> = () => {
+  const { name } = useParams();
+  const { namespace } = useWorkspaceInfo();
+
+  const [releasePlan, loaded, loadErr] = useK8sWatchResource<ReleasePlanKind>({
```
Comment (rohitkrai03) [inline src/pages/TriggerReleasePlanPage.tsx:21]: So I can see you are sending release plan name as a parameter in URL and then fetching the complete object here which is then passed on to the next component to use application name from it. Why can't you just send the application name directly in URL param and use it?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/pages/TriggerReleasePlanPage.tsx [hunk 2]
```diff
@@ -1,20 +1,66 @@
...
+    groupVersionKind: ReleasePlanGroupVersionKind,
+    isList: false,
+    name,
+    namespace,
+  });
+
   const accessReviewResources: AccessReviewResources = [{ model: ReleaseModel, verb: 'create' }];
 
+  if (!name) {
```
Comment (sahil143) [inline src/pages/TriggerReleasePlanPage.tsx:30]: you should check for the `releasePlan` here to see if it exists. If `name` is not present in URL, you should show the form without release plan preselected so the user can fill out the form.

**Review by sahil143 — CHANGES_REQUESTED** [review]

File: src/pages/TriggerReleasePlanPage.tsx (hunk 2 above)
Comment (abhinandan13jan) [reply src/pages/TriggerReleasePlanPage.tsx]: updated

File: src/pages/TriggerReleasePlanPage.tsx (hunk 1 above)
Comment (rohitkrai03) [reply src/pages/TriggerReleasePlanPage.tsx:21]: Do we actually need to check the releaseplan exist or not? Most of the use cases will be around triggering the release from the releaseplan action menu which means the plan actually exist. And how did the form work earlier without sending releaseplan as prop to `TriggerReleaseFormPage`

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Comment by scoheb** [comment]
Has this been deployed yet?

**Review by sahil143 — APPROVED** [review]

**Comment by scoheb** [comment]
Can you confirm if this has been promoted to stage please

**Pull request merged**